proposals/gc/Overview.md

```diff
@@ -1 +1,716 @@
+# GC Extension
+
+## Introduction
+
+### Motivation
+
+* Efficient support for high-level languages
+  - faster execution
+  - smaller deliverables
+  - the vast majority of modern languages need it
+
+* Efficient interoperability with embedder heap
+  - for example, DOM objects on the web
+  - no space leaks from mapping bidirectional tables
+
+* Provide access to industrial-strength GCs
+  - at least on the web, VMs already have high performance GCs
+
+* Non-goal: seamless interoperability between multiple languages
+
+
+### Challenges
+
+* Fast but type-safe
+* Lean but sufficiently universal
+* Language-independent
+* Trade-off triangle between simplicity, expressiveness and performance
+* Interaction with threads
+
+
+### Approach
+
+* Only basic but general structure: tuples (structs) and arrays.
+* No heavyweight object model.
+* Accept minimal amount of dynamic overhead (checked casts) as price for simplicity/universality.
+* Independent from linear memory.
+* Pay as you go.
+* Avoid generics or other complex type structure _if possible_.
+* Extend the design iteratively, ship a minimal set of functionality fast.
+
+
+### Requirements
+
+* Allocation of structures on the heap which are garbage collected.
```

Comment:
Conceptually, the (newly introduced) Wasm heap -- which, in a JS embedding, will happen to be shared with JS.